### DISCUSSION TITLE
How to use/define emotes

### DISCUSSION THREAD

**Discussion by MoriMoriMorii** [Monika-After-Story/MonikaModDev#9072] [general - submod / external content]
I want to write a submod that uses some new emotes I drew and I'm not exactly sure how to do this. I've never used the emote system before so I need a little help
Thanks!

**Comment by Booplicate** [comment]
You want to add new expressions? There's no direct support for that, so it'd be complex. You could start from reading [sprite-decoder](https://github.com/Monika-After-Story/MonikaModDev/blob/master/Monika%20After%20Story/game/sprite-decoder.rpy) code.

**Reply by MoriMoriMorii** [reply]
I was talking about the emote system (game/mod_assets/monika/f/"face-emote-confuse.png"), is it still able to be used? or was it deprecated.

**Reply by Booplicate** [reply]
Right now it's unused because it looks too game-y.

**Reply by MoriMoriMorii** [reply]
Right, is the system still in place?.

**Reply by Booplicate** [reply]
The system that allows to show different expression is, but emotes are disabled. For all available expressions use the [expression previewer](https://github.com/Monika-After-Story/MonikaModDev/wiki/FAQ#how-do-i-find-the-spritecode-for-an-expression).

**Reply by MoriMoriMorii** [reply]
I've uncommented the emote code (At least I'm pretty sure I did) and monika disappears when I use the emote, the file came with the game so I'm wondering what happened.

**Reply by MoriMoriMorii** [reply]
I'm also wondering how you define the emotes (For actual use) and where that code is.

**Reply by ThePotatoGuy** [reply]
As mentioned, the core code that shows emotes still works. However, the layer that takes the sprite code in text (i.e: `monika 1eua`) and calls the core code does **not** allow emotes. That being said, there's little reason for us to gatekeep that functionality, especially since adding new expressions can be a useful part for new submods. I'll create an issue to add the support for custom expressions and re-enable emote as a sprite code option.

There's a decent amount of code that would be changed, so I don't recommend attempting to do this for yourself unless you have coding experience. Basically you would need to modify sprite-decoder to parse emotes in `parse_exp_to_kwargs` into the `emote` kwarg.

**Reply by Booplicate** [reply]
> there's little reason for us to gatekeep that functionality, especially since adding new expressions can be a useful part for new submods. I'll create an issue to add the support for custom expressions and re-enable emote as a sprite code option.

There's a reason, actually. If we expose that part of API, we'd have to maintain it down the line. I don't think the emotes are a good way to express emotions. Facial expressions work much better.

**Reply by ThePotatoGuy** [reply]
**We** don't use emotes, but that doesn't mean submods should be restricted from using them. 

APIs for custom expressions is natural part of extending configurability. Right now devs can supply their own JSONs and images so we  actually already support this, but not in a way that won't trash the additions each update.

**Reply by Booplicate** [reply]
> We don't use emotes, but that doesn't mean submods should be restricted from using them.

Implementing X because it might be used by someone else (but not us) and then having to maintain it between versions doesn't sound right to me. Might end up being a redundant functionality which would be dragged across versions just to remain compatible. I don't think it's a good way of managing code, unless I misunderstand what you mean. A good way would be to allow register custom expressions (if it won't make expression codes more ugly/spaghetti, it's already hard to read).

So if a person wants some new expression (an emote, a new sweat, a new mouth), they implement it themselves using some exposed frameworks.
